src/shared-components/optionButton/style.ts

```diff
@@ -2,28 +2,43 @@ import styled from '@emotion/styled';
 import { css } from '@emotion/core';
 
 import { style as TYPOGRAPHY_STYLE } from '../typography';
-import { COLORS, ANIMATION, SPACER, BOX_SHADOWS } from '../../constants';
-import Container from '../container';
+import {
+  COLORS, ANIMATION, SPACER, BOX_SHADOWS, 
+} from '../../constants';
+import { containerStyles, ContainerType } from '../container/style';
 
-const getOptionTypeStyles = optionType => {
+import { OptionButtonProps } from '.';
+
+type BaseIconWrapperStylesProps = Required<
+  Pick<OptionButtonProps, 'buttonType' | 'optionType' | 'selected'>
+>;
+
+const getOptionTypeStyles = (
+  optionType: BaseIconWrapperStylesProps['optionType'],
+) => {
   if (optionType === 'checkbox') {
     return css`
       border-radius: 4px;
     `;
   }
+
   return css`
     border-radius: 50%;
   `;
 };
 
-const getTypeColor = type => {
-  if (type === 'secondary') {
+const getTypeColor = (buttonType: BaseIconWrapperStylesProps['buttonType']) => {
```

Comment:
This is cool - I didn't realize you could access properties off of a `type` like this.

Reply:
Yep--it's how you index into it. 

You probably won't ever need to do it but you could also do it to pull the type of an item in an array, by keying in with `[number]`:

<img width="370" alt="Screen Shot 2020-07-15 at 5 42 29 PM" src="https://user-images.githubusercontent.com/13544620/87607309-aad0ff00-c6c2-11ea-9075-dd994490aabe.png">